backend/infrahub/cli/patch.py

```diff
@@ -0,0 +1,142 @@
+
+    patch_module = importlib.import_module(patch_path)
+    patch_query_class = None
+    for _, cls in inspect.getmembers(patch_module, inspect.isclass):
+        if issubclass(cls, PatchQuery) and cls is not PatchQuery:
+            patch_query_class = cls
+            break
+    if patch_query_class is None:
+        rprint(f"{ERROR_BADGE} No subclass of infrahub.patch.queries.base.PatchQuery found in {patch_path}")
+        raise typer.Exit(1)
+
+    config.load_and_exit(config_file_name=config_file)
+
+    context: CliContext = ctx.obj
+    db = await context.init_db(retry=1)
+
+    patch_query_instance = patch_query_class(db=db)
+    patch_runner = get_patch_runner(db=db)
+    patch_plan_dir = await patch_runner.prepare_plan(patch_query_instance, directory=Path(patch_plans_dir))
+    rprint(f"{SUCCESS_BADGE} Patch plan created at {patch_plan_dir}")
+
+    await db.close()
+
+
+@patch_app.command(name="apply")
+async def apply_patch_cmd(
+    ctx: typer.Context,
+    patch_plan_dir: Path = typer.Argument(help="Path to the directory containing a patch plan"),
+    config_file: str = typer.Argument("infrahub.toml", envvar="INFRAHUB_CONFIG"),
+) -> None:
+    """Apply a given patch plan"""
+    logging.getLogger("infrahub").setLevel(logging.WARNING)
+    logging.getLogger("neo4j").setLevel(logging.ERROR)
+    logging.getLogger("prefect").setLevel(logging.ERROR)
+
+    config.load_and_exit(config_file_name=config_file)
+
+    context: CliContext = ctx.obj
+    dbdriver = await context.init_db(retry=1)
+
+    if not patch_plan_dir.exists() or not patch_plan_dir.is_dir():
+        rprint(f"{ERROR_BADGE} patch_plan_dir must be an existing directory")
+        raise typer.Exit(1)
+
+    async with dbdriver.start_session() as db:
+        patch_runner = get_patch_runner(db=db)
+        await patch_runner.apply(patch_plan_directory=patch_plan_dir)
+        rprint(f"{SUCCESS_BADGE} Patch plan successfully applied")
+
+    await dbdriver.close()
+
+
+@patch_app.command(name="revert")
+async def revert_patch_cmd(
+    ctx: typer.Context,
+    patch_plan_dir: Path = typer.Argument(help="Path to the directory containing a patch plan"),
+    config_file: str = typer.Argument("infrahub.toml", envvar="INFRAHUB_CONFIG"),
+) -> None:
+    """Revert a given patch plan"""
+    logging.getLogger("infrahub").setLevel(logging.WARNING)
+    logging.getLogger("neo4j").setLevel(logging.ERROR)
+    logging.getLogger("prefect").setLevel(logging.ERROR)
+    config.load_and_exit(config_file_name=config_file)
+
+    context: CliContext = ctx.obj
+    db = await context.init_db(retry=1)
+
+    if not patch_plan_dir.exists() or not patch_plan_dir.is_dir():
+        rprint(f"{ERROR_BADGE} patch_plan_dir must be an existing directory")
+        raise typer.Exit(1)
+
+    patch_runner = get_patch_runner(db=db)
+    await patch_runner.revert(patch_plan_directory=patch_plan_dir)
+    rprint(f"{SUCCESS_BADGE} Patch plan successfully reverted")
+
+    await db.close()
```

Comment:
It could be nice to have a `plan_and_apply_patch` command to make it simpler to use?